OrbitCore/LinuxEventTracer.h

```diff
@@ -7,25 +7,39 @@
 //-----------------------------------
 #pragma once
 
+#include <atomic>
+#include <functional>
 #include <thread>
 
 //-----------------------------------------------------------------------------
 class LinuxEventTracer
 {
 public:
     static const uint32_t ROUND_ROBIN_BATCH_SIZE = 5;
-    typedef std::function<void(const std::string& a_Data)> Callback;
+    using Callback = std::function<void(const std::string& a_Data)>;
+
+    explicit LinuxEventTracer(pid_t a_Pid, uint32_t a_Frequency = 1000) :
+            m_Pid{a_Pid}, m_Frequency{a_Frequency} {}
+
     void Start();
     void Stop();
-    bool IsRunning() const { return !m_ExitRequested; }
+    bool IsRunning() const { return !(*m_ExitRequested); }
 
 protected:
     void CommandCallback(std::string a_Line);
 
 private:
-    bool m_ExitRequested = true;
+    pid_t m_Pid;
+    uint32_t m_Frequency;
```

Comment:
Could there be a case where I want frequency to be smaller than 1? Sample every 5 seconds, for example?

Reply:
For the moment, I'm not sure. I can switch to using the period or leave it as is as it's easy to change in the future.

Reply:
The way it is used here, frequency is passed directly to perf_event_open, with a flag saying that we are passing a frequency and not a period.